api/v1beta1/tags.go

```diff
@@ -95,11 +99,37 @@ func (t Tags) Validate() []*field.Error {
 				field.Invalid(field.NewPath("spec", "additionalTags"), v, "value cannot be longer than 256 characters"),
 			)
 		}
+		if isUserOwnedTag(k) {
+			userTagCount++
+		}
+		val := re.MatchString(k)
+		if !val {
+			errs = append(errs,
+				field.Invalid(field.NewPath("spec", "additionalTags"), k, "key cannot have characters other than alphabets, numbers, spaces and _ . : / = + - @ ."),
+			)
+		}
+		val = re.MatchString(v)
+		if !val {
+			errs = append(errs,
+				field.Invalid(field.NewPath("spec", "additionalTags"), v, "value cannot have characters other than alphabets, numbers, spaces and _ . : / = + - @ ."),
+			)
+		}
+	}
+
+	if userTagCount > maxUserTagsAllowed {
+		errs = append(errs,
+			field.Invalid(field.NewPath("spec", "additionalTags"), t, "user created tags cannot be more than 50"),
+		)
 	}
 
 	return errs
 }
 
+// Checks whether the tag created is user tag or not.
+func isUserOwnedTag(k string) bool {
+	return len(k) > 3 && k[0:4] != "aws:"
```

Comment:
Users cannot create tags start with `aws:`, this check is not needed. But we can add a check to disallow adding tags  with `aws:` prefix.

Reply:
I thought the list of tags we get to validate contained both user and system tags.

I tried to calculate the number of user tags we create because the upper limit is 50 for that. Thus having a check to know whether the tag is system created or not for counting purposes was important.

This is what I thought :))

what does everyone suggest ??

Reply:
Can I get some clearance here please @richardcase @sedefsavas 
Kinda confused here :))

Reply:
additionalTags that we use this method for do not include system tags.
We can add a check to disallow tags with system prefix in this validation method.
I tried manually, AWS already errors when user tag name starts with `aws:`

Since additional tags are not supposed to have system tags, we don't need to check if a tag is a system tag here for the limit calculation.

Reply:
cool thanks understood :))
Will soon commit changes.